### ISSUE TITLE
No more Catalina compatibility?

### ISSUE THREAD

**Issue by esmukherjee** [bermudalocket/redditweaks#15]
Releases past 1.6.1 cannot be installed on my machine running 10.15.7. Was this intended behavior in the rewrite? 

I'm stuck with Catalina for a while as Big Sur is not an approved OS for my workplace. Thanks for all your hard work on this extension.

**Comment by bermudalocket** [comment]
Yeah, Catalina compat was lost with the rewrite. I've had multiple requests to bring it back, though, and have added it to the to-do list for the next release.

**Comment by bermudalocket** [comment]
I have to defer this beyond next release as it's more complicated than I thought it would be. Notably, Catalina has no equivalent to `@StateObject`, `@AppStorage`, `Menu`, `Image(systemName:)`, `Image.imageScale(_ scale:)`, and `ProgressView`. Working around `ProgressView` is trivial; working around the missing image stuff is annoying (having to manually export every SFSymbol I've used and add them as assets) but still trivial; and working around `Menu` would mean having to create a custom UI control. Those are all doable. The major problems are the lack of `@StateObject` and `@AppStorage`, which are critical to the flow of data through the view hierarchy.

I can make this happen, but it won't happen quickly. Designing, implementing, and testing Catalina-friendly home-cooked versions of the aforementioned property wrappers is most likely something I'll chip away at in my free time.

**Comment by esmukherjee** [comment]
Much appreciation that you're even looking into it! And 1.6.1 is already great. Makes Safari usable on reddit and you don't miss RES at all.
